### PR TITLE
chore : Update README.md to have table of contents

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

See preview [here](https://github.com/rohanKanojia/jkube/blob/pr/restructure-readme/README.md).

- Update README.md to have table of contents with useful links
- Add Kubernetes Gradle Plugin and OpenShift Gradle Plugin sections
- Add YouTube video demo links to plugin sections
- Remove JKube Kit section from README, at the moment I don't think anyone would be directly using this. Nor users would be interested in the project internal structure.

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] My code follows the style guidelines of this project
 - [X] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [X] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [X] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Review by manusa — APPROVED** [review]
It looks fine.
Note that GitHub creates the TOC automatically as a button for every README markdown file. However, I guess this isn't still the most usable option for users and many don't even know about it.

File: README.md [hunk 1]
```diff
@@ -35,43 +46,65 @@ This project contains various building blocks for the Kubernetes Java developer
...
+- [![Maven Central](https://img.shields.io/maven-central/v/org.eclipse.jkube/kubernetes-maven-plugin.svg?label=Maven%20Central)](https://search.maven.org/search?q=g:%22org.eclipse.jkube%22%20AND%20a:%22kubernetes-maven-plugin%22)
+- [![Documentation](https://img.shields.io/badge/plugin-documentation-lightgrey)](https://www.eclipse.org/jkube/docs/kubernetes-maven-plugin)
+- Add to project:
+  ```xml
+  <plugin>
+    <groupId>org.eclipse.jkube</groupId>
+    <artifactId>kubernetes-maven-plugin</artifactId>
+    <version>${jkube.version}</version>
+  </plugin>
+  ```
```
Comment (sunix) [inline README.md:58]: Maybe add an example after:
```suggestion
   - Build, create and push the container image,  generate and apply the Kubernetes resources
   ```
   mvn package k8s:build k8s:push k8s:generate k8s:apply
   ```
```

File: README.md [hunk 2]
```diff
@@ -35,43 +46,65 @@ This project contains various building blocks for the Kubernetes Java developer
...
+- :tv: Watch 2-minute demo on YouTube:
```
Comment (sunix) [inline README.md:59]: Maybe provide an example of what's next ?

```suggestion

    - Build, create and push the container image,  generate and apply the Kubernetes resources
      ```
      mvn package k8s:build k8s:push k8s:generate k8s:apply
      ```

- :tv: Watch 2-minute demo on YouTube:
```

Comment (sunix) [reply README.md:59]: No I mean after each plugins, provide some example of how to use the plugin.

Comment (rohanKanojia) [reply README.md:63]: I've added it as per your suggestion

**Review by sunix — APPROVED** [review]

**Pull request merged**